src/os/bluestore/aio.cc

```diff
@@ -15,6 +25,50 @@ std::ostream& operator<<(std::ostream& os, const aio_t& aio)
   return os;
 }
 
+#if defined(__FreeBSD__)
+std::ostream& operator<<(std::ostream& os, const struct aiocb *cb)
+{
+  os << "aiocb @ "    << &cb << "(";
+  os << "fildes = "   << cb->aio_fildes;
+  os << ", offset = " << cb->aio_offset;
+  os << ", nbytes = " << cb->aio_nbytes;
+  os << ", buf = 0x"    << std::hex << (size_t)cb->aio_buf << std::dec;
+  os << ", lio_opcode = " << cb->aio_lio_opcode;
+  os << ", priv.status = " << cb->_aiocb_private.status;
+  os << ", priv.error = " << cb->_aiocb_private.error;
+  os << ", priv.sigevent.notify = " << cb->aio_sigevent.sigev_notify;
+  os << ", priv.sigevent.notify_kqueue = " << cb->aio_sigevent.sigev_notify_kqueue;
+  os << ", priv.sigevent.kevent_flags = " << cb->aio_sigevent.sigev_notify_kevent_flags;
+  os << ")";
+  return os;
+}
+#endif
+
+int aio_queue_t::init(int f) {
```

Comment:
please split this change into functional changes and non-functional changes to help reviewer to understand your changes. in this case, you could have two changes: one to move the init() to .cc file. another to improve it. it's difficult to me to track the changes.

Reply:
@tchaikov 
Perhaps better leaving the complete body in the headerfile then?

Reply:
@tchaikov 
Moved the init() function back into the headerfile.

Are there more things to do before you can take this PR for merging?